wrappers/services.go

```diff
@@ -582,9 +582,42 @@ func (es *ensureSnapServicesContext) reloadModified() error {
 	return nil
 }
 
+// quotaGroupByName is a helper to find the quota group by name in a quota group slice.
+func quotaGroupByName(name string, allGrps []*quota.Group) *quota.Group {
+	for _, grp := range allGrps {
+		if grp.Name == name {
+			return grp
+		}
+	}
+	return nil
+}
+
+// resolveAppSpecificQuotaGroup takes into account any service-specific quota group. Quota groups
+// are assigned on snap basis in general, which means the quota applies to everything inside the snap, unless
+// a sub-group exists that refer to a specific service. This function checks for that, and returns the correct
+// quota group, should a sub-group exist for this specific appInfo.
+func resolveAppSpecificQuotaGroup(appInfo *snap.AppInfo, snapQuotaGroup *quota.Group, allGrps []*quota.Group) *quota.Group {
```

Comment:
two things, I think we have a check that we don't put the same snap in more than one group? do we have the same check for snap services, that they don't end up in more than one group? we should

Reply:
I agree with your comments on the approach, I'll refactor the loop approach a bit